internal/events/types/hype_train/hype_train_begin_event.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package hype_train_begin
+
+import(
+	"encoding/json"
+	"time"
+
+	"github.com/twitchdev/twitch-cli/internal/events"
+	"github.com/twitchdev/twitch-cli/internal/models"
+	"github.com/twitchdev/twitch-cli/internal/util"
+)
+
+var transportsSupported = map[string]bool{
+	models.TransportWebSub:   false,
+	models.TransportEventSub: true,
+}
+
+var triggerSupported = []string{"hype-train-begin"}
+
+var triggerMapping = map[string]map[string]string{
+	models.TransportEventSub: {
+		"hype-train-begin":  "channel.hype_train.begin",
+	},
+}
+
+type Event struct{}
+
+func (e Event) GenerateEvent(params events.MockEventParameters) (events.MockEventResponse, error) {
+	var event []byte
+	var err error
+
+	//Some values
+	StartedAtTime := util.GetTimestamp()
+	localTotal := util.RandomViewerCount()
+	localGoal := util.RandomViewerCount()
+	localProgress := (localTotal/localGoal)
+	localRandomUser1 := util.RandomUserID()
+	localRandomUser2 := util.RandomUserID()
+	localRandomUser3 := util.RandomUserID()
+	localLC := models.ContributionData{TotalContribution: util.RandomViewerCount(), TypeOfContribution: util.RandomType(), UserWhoMadeContribution: localRandomUser1, UserNameWhoMadeContribution: localRandomUser1, UserLoginWhoMadeContribution: localRandomUser1}
+	localTC := []models.ContributionData{{TotalContribution: util.RandomViewerCount(), TypeOfContribution: util.RandomType(), UserWhoMadeContribution: localRandomUser2, UserNameWhoMadeContribution: localRandomUser2, UserLoginWhoMadeContribution: localRandomUser2},{TotalContribution: util.RandomViewerCount(), TypeOfContribution: util.RandomType(), UserWhoMadeContribution: localRandomUser3, UserNameWhoMadeContribution: localRandomUser3, UserLoginWhoMadeContribution: localRandomUser3}}
+	ExpiresAtTime := util.GetTimestamp()
```

Comment:
Would definitely recommend making these inline whenever possible; little easier to read.